### ISSUE TITLE
Missing upgrade instructions (5.5) concerning Tinker's ClassAliasAutoloader?

### ISSUE THREAD

**Issue by mortenscheel** [laravel/internals#841]
The auto-aliasing functionality of spatie/laravel-tinker-tools has been included in the current version of Tinker, and it works great on fresh Laravel 5.5 projects.

However it didn't work on any of the older projects I've upgraded to 5.5.
Turns out I needed to add "optimize-autoloader": true to the config key in composer.json to make it work.

Might be worth adding to the upgrade instructions.

**Comment by Dylan-DPC** [comment]
you can send a PR which does that :smiley:

**Comment by sisve** [comment]
> Note: You should not enable any of these optimizations in development as they all will cause various problems when adding/removing classes. The performance gains are not worth the trouble in a development setting.

Source: https://getcomposer.org/doc/articles/autoloader-optimization.md

I believe that what really fixed your issue was something else you did when trying to debug your issue, not the optimized autoloader.

**Comment by mortenscheel** [comment]
@sisve The optimizations are enabled by default i 5.5. It's true that added/removed classes won't work until you've run a `composer dump-autoload` (or any other composer command that triggers dumping new autoload files), but I think that's all there is to it.

When I investigated the issue, I noticed that the autoload file structure was different in my upgraded projects compared to a fresh 5.5 install. Adding "optimize-autoloader": true and running `composer dump-autoload` made everything work as expected. I really can't see what else could have fixed it.
